tests/lib/sprintf/src/main.c

```diff
@@ -764,8 +778,10 @@ void test_fprintf(void)
 	ret = fprintf(stdout, "");
 	zassert_equal(ret, 0, "fprintf failed!");
 
+#ifndef __PICOLIBC__	/* this is UB */
```

Comment:
Hmm, we are checking against `CONFIG_PICOLIBC` in one place and against `__PICOLIBC__` in other places in same file. I'm wondering if it is intended?

Reply:
This patch is part of PR #44096, and I've fixed that to use CONFIG_PICOLIBC consistently.